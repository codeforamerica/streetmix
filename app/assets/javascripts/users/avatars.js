var avatarCache = {};

function _updateAvatars() {
  var els = document.querySelectorAll('.avatar:not([loaded])');

  for (var i = 0, el; el = els[i]; i++) {
    var twitterId = el.getAttribute('twitterId');

    if (avatarCache[twitterId]) {
      el.style.backgroundImage = 'url(' + avatarCache[twitterId] + ')';
      el.setAttribute('loaded', true);
    }
  }
}

function _fetchAvatars() {

  // NOTE:
  // This function might be called on very old browsers. Please make
  // sure not to use modern faculties.

  if (typeof document.querySelectorAll == 'undefined') {
    return;
  }

  var els = document.querySelectorAll('.avatar:not([loaded])');

  for (var i = 0, el; el = els[i]; i++) {
    var twitterId = el.getAttribute('twitterId');
    var postpone = el.getAttribute('postpone');

    if (twitterId && !postpone && (typeof avatarCache[twitterId] == 'undefined')) {
      _fetchAvatar(twitterId);
    }
  }

  _updateAvatars();
}

function _fetchAvatar(twitterId) {
  avatarCache[twitterId] = null;

  $.ajax({
    dataType: 'json',
    url: API_URL + 'v2/users/' + twitterId
  }).done(_receiveAvatar);
}

function _receiveAvatar(details) {
  if (details && details.twitterId && details.twitterProfileImageUrl) {
    avatarCache[details.twitterId] = details.twitterProfileImageUrl;
    _updateAvatars();
  }
}
