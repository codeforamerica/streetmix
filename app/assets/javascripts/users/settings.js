var TRACK_ACTION_ERROR_RM2 = 'Error RM2 (auth 401 failure mid-flight)';

var LOCAL_STORAGE_SETTINGS_ID = 'settings';
var LOCAL_STORAGE_SETTINGS_WELCOME_DISMISSED = 'settings-welcome-dismissed';
var LOCAL_STORAGE_SIGN_IN_ID = 'sign-in';
var LOCAL_STORAGE_FEEDBACK_BACKUP = 'feedback-backup';
var LOCAL_STORAGE_FEEDBACK_EMAIL_BACKUP = 'feedback-email-backup';
var SAVE_SETTINGS_DELAY = 500;

var settings = {
  lastStreetId: null,
  lastStreetNamespacedId: null,
  lastStreetUserId: null,
  priorLastStreetId: null, // Do not save
  newStreetPreference: null,

  saveAsImageTransparentSky: null,
  saveAsImageSegmentNamesAndWidths: null,
  saveAsImageStreetName: null
};
var settingsWelcomeDismissed = false;

var saveSettingsTimerId = -1;

var signedIn = false;
var signInLoaded = false;
var signInData = null;

function _mergeAndFillDefaultSettings(secondSettings) {
  // Merge with local settings

  if (!settings.newStreetPreference) {
    settings.newStreetPreference = secondSettings.newStreetPreference;
  }
  if (typeof settings.lastStreetId === 'undefined') {
    settings.lastStreetId = secondSettings.lastStreetId;
  }
  if (typeof settings.lastStreetNamespacedId === 'undefined') {
    settings.lastStreetNamespacedId = secondSettings.lastStreetNamespacedId;
  }
  if (typeof settings.lastStreetCreatorId === 'undefined') {
    settings.lastStreetCreatorId = secondSettings.lastStreetCreatorId;
  }
  if (typeof settings.saveAsImageTransparentSky === 'undefined') {
    settings.saveAsImageTransparentSky = secondSettings.saveAsImageTransparentSky;
  }
  if (typeof settings.saveAsImageSegmentNamesAndWidths === 'undefined') {
    settings.saveAsImageSegmentNamesAndWidths = secondSettings.saveAsImageSegmentNamesAndWidths;
  }
  if (typeof settings.saveAsImageStreetName === 'undefined') {
    settings.saveAsImageStreetName = secondSettings.saveAsImageStreetName;
  }

  // Provide defaults if the above failed

  if (!settings.newStreetPreference) {
    settings.newStreetPreference = NEW_STREET_DEFAULT;
  }
  if (typeof settings.lastStreetId === 'undefined') {
    settings.lastStreetId = null;
  }
  if (typeof settings.lastStreetNamespacedId === 'undefined') {
    settings.lastStreetNamespacedId = null;
  }
  if (typeof settings.lastStreetCreatorId === 'undefined') {
    settings.lastStreetCreatorId = null;
  }
  if (typeof settings.saveAsImageTransparentSky === 'undefined') {
    settings.saveAsImageTransparentSky = false;
  }
  if (typeof settings.saveAsImageSegmentNamesAndWidths === 'undefined') {
    settings.saveAsImageSegmentNamesAndWidths = false;
  }
  if (typeof settings.saveAsImageStreetName === 'undefined') {
    settings.saveAsImageStreetName = false;
  }
}

function _loadSettingsWelcomeDismissed() {
  if (window.localStorage[LOCAL_STORAGE_SETTINGS_WELCOME_DISMISSED]) {
    settingsWelcomeDismissed =
        JSON.parse(window.localStorage[LOCAL_STORAGE_SETTINGS_WELCOME_DISMISSED]);
  }
}

function _saveSettingsWelcomeDismissed() {
  window.localStorage[LOCAL_STORAGE_SETTINGS_WELCOME_DISMISSED] =
      JSON.stringify(settingsWelcomeDismissed);
}

function _loadSettings() {
  if (signedIn && signInData.details) {
    var serverSettings = signInData.details.data;
  } else {
    var serverSettings = {};
  }

  // TODO handle better if corrupted
  if (window.localStorage[LOCAL_STORAGE_SETTINGS_ID]) {
    var localSettings = JSON.parse(window.localStorage[LOCAL_STORAGE_SETTINGS_ID]);
  } else {
    var localSettings = {};
  }

  settings = {};

  if (serverSettings) {
    settings = serverSettings;
  }
  _mergeAndFillDefaultSettings(localSettings);

  if (mode == MODES.JUST_SIGNED_IN) {
    settings.lastStreetId = localSettings.lastStreetId;
    settings.lastStreetNamespacedId = localSettings.lastStreetNamespacedId;
    settings.lastStreetCreatorId = localSettings.lastStreetCreatorId;
  }

  settings.priorLastStreetId = settings.lastStreetId;

  _saveSettingsLocally();
}

function _trimSettings() {
  var data = {};

  data.lastStreetId = settings.lastStreetId;
  data.lastStreetNamespacedId = settings.lastStreetNamespacedId;
  data.lastStreetCreatorId = settings.lastStreetCreatorId;
  data.saveAsImageTransparentSky = settings.saveAsImageTransparentSky;
  data.saveAsImageSegmentNamesAndWidths = settings.saveAsImageSegmentNamesAndWidths;
  data.saveAsImageStreetName = settings.saveAsImageStreetName;

  data.newStreetPreference = settings.newStreetPreference;

  return data;
}

function _saveSettingsLocally() {
  window.localStorage[LOCAL_STORAGE_SETTINGS_ID] =
      JSON.stringify(_trimSettings());

  _scheduleSavingSettingsToServer();
}

function _confirmSaveStreetToServerInitial() {
  saveStreetIncomplete = false;

  serverContacted = true;
  _checkIfEverythingIsLoaded();
}


function _saveSettingsToServer() {
  if (!signedIn || abortEverything) {
    return;
  }

  var transmission = JSON.stringify({ data: _trimSettings() });

  _newNonblockingAjaxRequest({
    // TODO const
    url: API_URL + 'v2/users/' + signInData.twitterId,
    data: transmission,
    dataType: 'json',
    type: 'PUT',
    contentType: 'application/json',
    headers: { 'Authorization': _getAuthHeader() }
  }, true, null, _errorSavingSettingsToServer);
}

function _errorSavingSettingsToServer(data) {
  if (!abortEverything && (data.status == 401)) {
    Stmx.app.eventTracking.track(TRACK_CATEGORY_ERROR, TRACK_ACTION_ERROR_RM2,
        null, null, false);

    mode = MODES.FORCE_RELOAD_SIGN_OUT_401;
    _processMode();
  }
}

function _scheduleSavingSettingsToServer() {
  if (!signedIn) {
    return;
  }

  _clearScheduledSavingSettingsToServer();

  saveSettingsTimerId =
      window.setTimeout(function() { _saveSettingsToServer(); }, SAVE_SETTINGS_DELAY);
}

function _clearScheduledSavingSettingsToServer() {
  window.clearTimeout(saveSettingsTimerId);
}

function _onStorageChange() {
  if (signedIn && !window.localStorage[LOCAL_STORAGE_SIGN_IN_ID]) {
    mode = MODES.FORCE_RELOAD_SIGN_OUT;
    _processMode();
  } else if (!signedIn && window.localStorage[LOCAL_STORAGE_SIGN_IN_ID]) {
    mode = MODES.FORCE_RELOAD_SIGN_IN;
    _processMode();
  }
}
