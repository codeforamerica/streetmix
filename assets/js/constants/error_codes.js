var ERRORS = {
  NOT_FOUND: 1,
  SIGN_OUT: 2,
  NO_STREET: 3, // for gallery if you delete the street you were looking at
  FORCE_RELOAD_SIGN_IN: 4,
  FORCE_RELOAD_SIGN_OUT: 5,
  STREET_DELETED_ELSEWHERE: 6,
  NEW_STREET_SERVER_FAILURE: 7,
  FORCE_RELOAD_SIGN_OUT_401: 8,
  TWITTER_ACCESS_DENIED: 9,
  AUTH_PROBLEM_NO_TWITTER_REQUEST_TOKEN: 10,
  AUTH_PROBLEM_NO_TWITTER_ACCESS_TOKEN: 11,
  AUTH_PROBLEM_API_PROBLEM: 12,
  GENERIC_ERROR: 13,
  UNSUPPORTED_BROWSER: 14,
  STREET_404: 15,
  STREET_404_BUT_LINK_TO_USER: 16,
  STREET_410_BUT_LINK_TO_USER: 17,
  CANNOT_CREATE_NEW_STREET_ON_PHONE: 18,
  SIGN_IN_SERVER_FAILURE: 19,
  SIGN_IN_401: 20
};
