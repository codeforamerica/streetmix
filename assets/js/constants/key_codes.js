var KEYS = {
  LEFT_ARROW: 37,
  RIGHT_ARROW: 39,
  ENTER: 13,
  BACKSPACE: 8,
  DELETE: 46,
  ESC: 27,
  D: 68,
  S: 83,
  Y: 89,
  Z: 90,
  EQUAL: 187, // = or +
  EQUAL_ALT: 61, // Firefox
  PLUS_KEYPAD: 107,
  MINUS: 189,
  MINUS_ALT: 173, // Firefox
  MINUS_KEYPAD: 109,
  SLASH: 191 // slash or question mark
};
