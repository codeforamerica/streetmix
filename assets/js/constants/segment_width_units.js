var IMPERIAL_METRIC_MULTIPLIER = 30 / 100;
var COUNTRIES_IMPERIAL_UNITS = ['US'];

var WIDTH_INPUT_CONVERSION = [
  { text: 'm', multiplier: 1 / IMPERIAL_METRIC_MULTIPLIER },
  { text: 'cm', multiplier: 1 / 100 / IMPERIAL_METRIC_MULTIPLIER },
  { text: '"', multiplier: 1 / 12 },
  { text: 'inch', multiplier: 1 / 12 },
  { text: 'inches', multiplier: 1 / 12 },
  { text: '\'', multiplier: 1 },
  { text: 'ft', multiplier: 1 },
  { text: 'feet', multiplier: 1 }
];

var SEGMENT_WIDTH_RESOLUTION_IMPERIAL = .25;
var SEGMENT_WIDTH_CLICK_INCREMENT_IMPERIAL = .5;
var SEGMENT_WIDTH_DRAGGING_RESOLUTION_IMPERIAL = .5;

// don't use const because of rounding problems
var SEGMENT_WIDTH_RESOLUTION_METRIC = 1 / 3; // .1 / IMPERIAL_METRIC_MULTIPLER
var SEGMENT_WIDTH_CLICK_INCREMENT_METRIC = 2 / 3; // .2 / IMPERIAL_METRIC_MULTIPLER
var SEGMENT_WIDTH_DRAGGING_RESOLUTION_METRIC = 2 / 3; // .2 / IMPERIAL_METRIC_MULTIPLER

var MIN_WIDTH_EDIT_CANVAS_WIDTH = 120;
var WIDTH_EDIT_MARGIN = 20;

var NORMALIZE_PRECISION = 5;
var METRIC_PRECISION = 3;
var WIDTH_ROUNDING = .01;

var SEGMENT_WARNING_OUTSIDE = 1;
var SEGMENT_WARNING_WIDTH_TOO_SMALL = 2;
var SEGMENT_WARNING_WIDTH_TOO_LARGE = 3;

var PRETTIFY_WIDTH_OUTPUT_MARKUP = 1;
var PRETTIFY_WIDTH_OUTPUT_NO_MARKUP = 2;
var PRETTIFY_WIDTH_INPUT = 3;

var SETTINGS_UNITS_IMPERIAL = 1;
var SETTINGS_UNITS_METRIC = 2;

var IMPERIAL_VULGAR_FRACTIONS = {
  '.125': '⅛',
  '.25': '¼',
  '.375': '⅜',
  '.5': '½',
  '.625': '⅝',
  '.75': '¾',
  '.875': '⅞'
};