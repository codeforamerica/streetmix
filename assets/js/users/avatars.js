var avatarCache = {};

function _updateAvatars() {
  var els = document.querySelectorAll('.avatar:not([loaded])');

  for (var i = 0, el; el = els[i]; i++) {
    var userId = el.getAttribute('userId');

    if (avatarCache[userId]) {
      el.style.backgroundImage = 'url(' + avatarCache[userId] + ')';
      el.setAttribute('loaded', true);
    }
  }
}

function _fetchAvatars() {

  // NOTE:
  // This function might be called on very old browsers. Please make
  // sure not to use modern faculties.

  if (typeof document.querySelectorAll == 'undefined') {
    return;
  }

  var els = document.querySelectorAll('.avatar:not([loaded])');

  for (var i = 0, el; el = els[i]; i++) {
    var userId = el.getAttribute('userId');
    var postpone = el.getAttribute('postpone');

    if (userId && !postpone && (typeof avatarCache[userId] == 'undefined')) {
      _fetchAvatar(userId);
    }
  }

  _updateAvatars();
}

function _fetchAvatar(userId) {
  avatarCache[userId] = null;

  $.ajax({
    dataType: 'json',
    url: API_URL + 'v1/users/' + userId
  }).done(_receiveAvatar);
}

function _receiveAvatar(details) {
  if (details && details.id && details.profileImageUrl) {
    avatarCache[details.id] = details.profileImageUrl;
    _updateAvatars();
  }
}
