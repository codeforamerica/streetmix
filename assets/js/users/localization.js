var IP_GEOLOCATION_API_URL = 'https://freegeoip.net/json/'
var IP_GEOLOCATION_TIMEOUT = 1000; // After this time, we donâ€™t wait any more
var geolocationLoaded

var SETTINGS_UNITS_IMPERIAL = 1
var SETTINGS_UNITS_METRIC = 2

var units = SETTINGS_UNITS_IMPERIAL

var SEGMENT_WIDTH_RESOLUTION_IMPERIAL = .25
var SEGMENT_WIDTH_CLICK_INCREMENT_IMPERIAL = .5
var SEGMENT_WIDTH_DRAGGING_RESOLUTION_IMPERIAL = .5

// don't use const because of rounding problems
var SEGMENT_WIDTH_RESOLUTION_METRIC = 1 / 3; // .1 / IMPERIAL_METRIC_MULTIPLER
var SEGMENT_WIDTH_CLICK_INCREMENT_METRIC = 2 / 3; // .2 / IMPERIAL_METRIC_MULTIPLER
var SEGMENT_WIDTH_DRAGGING_RESOLUTION_METRIC = 2 / 3; // .2 / IMPERIAL_METRIC_MULTIPLER

var COUNTRIES_IMPERIAL_UNITS = ['US']

var leftHandTraffic = false

var COUNTRIES_LEFT_HAND_TRAFFIC =
['AI', 'AG', 'AU', 'BS', 'BD', 'BB', 'BM', 'BT', 'BW', 'BN',
  'KY', 'CX', 'CC', 'CK', 'CY', 'DM', 'TL', 'FK', 'FJ', 'GD', 'GG',
  'GY', 'HK', 'IN', 'ID', 'IE', 'IM', 'JM', 'JP', 'JE', 'KE', 'KI',
  'LS', 'MO', 'MW', 'MY', 'MV', 'MT', 'MU', 'MS', 'MZ', 'NA', 'NR',
  'NP', 'NZ', 'NU', 'NF', 'PK', 'PG', 'PN', 'SH', 'KN', 'LC', 'VC',
  'WS', 'SC', 'SG', 'SB', 'ZA', 'LK', 'SR', 'SZ', 'TZ', 'TH', 'TK',
  'TO', 'TT', 'TC', 'TV', 'UG', 'GB', 'VG', 'VI', 'ZM', 'ZW']

function _checkIfSignInAndGeolocationLoaded () {
  if (geolocationLoaded && signInLoaded) {
    switch (mode) {
      case MODES.NEW_STREET:
      case MODES.NEW_STREET_COPY_LAST:
        if (app.readOnly) {
          _showError(ERRORS.CANNOT_CREATE_NEW_STREET_ON_PHONE, true)
        } else {
          _createNewStreetOnServer()
        }
        break
    }
  }
}

function _detectGeolocation () {
  geolocationLoaded = false

  window.fetch(IP_GEOLOCATION_API_URL)
    .then(function (response) {
      return response.json()
    })
    .then(_receiveGeolocation)
    .catch(function (error) {
      console.log('_detectGeolocation', error)
    })

  window.setTimeout(_detectGeolocationTimeout, IP_GEOLOCATION_TIMEOUT)
}

function _detectGeolocationTimeout () {
  if (!geolocationLoaded) {
    geolocationLoaded = true
    document.querySelector('#loading-progress').value++
    _checkIfSignInAndGeolocationLoaded()
    _checkIfEverythingIsLoaded()

    trackEvent('ERROR', 'ERROR_GEOLOCATION_TIMEOUT', null, null, false)
  }
}

function _updateSettingsFromCountryCode (countryCode) {
  if (COUNTRIES_IMPERIAL_UNITS.indexOf(countryCode) != -1) {
    units = SETTINGS_UNITS_IMPERIAL
  } else {
    units = SETTINGS_UNITS_METRIC
  }

  if (COUNTRIES_LEFT_HAND_TRAFFIC.indexOf(countryCode) != -1) {
    leftHandTraffic = true
  }

  if (debug.forceLeftHandTraffic) {
    leftHandTraffic = true
  }
  if (debug.forceMetric) {
    units = SETTINGS_UNITS_METRIC
  }
}

function _receiveGeolocation (info) {
  if (geolocationLoaded) {
    // Timed out, discard results
    return
  }

  if (info && info.country_code) {
    _updateSettingsFromCountryCode(info.country_code)
  }
  if (info && info.ip) {
    system.ipAddress = info.ip
  }

  geolocationLoaded = true
  document.querySelector('#loading-progress').value++
  _checkIfSignInAndGeolocationLoaded()
  _checkIfEverythingIsLoaded()
}

function _updateUnits (newUnits) {
  if (street.units == newUnits) {
    return
  }

  units = newUnits
  street.units = newUnits

  // If the user converts and then straight converts back, we just reach
  // to undo stack instead of double conversion (which could be lossy).
  if (undoStack[undoPosition - 1] &&
    (undoStack[undoPosition - 1].units == newUnits)) {
    var fromUndo = true
  } else {
    var fromUndo = false
  }

  _propagateUnits()

  ignoreStreetChanges = true
  if (!fromUndo) {
    _normalizeAllSegmentWidths()

    if (street.remainingWidth == 0) {
      street.width = 0
      for (var i in street.segments) {
        street.width += street.segments[i].width
      }
    } else {
      street.width = _normalizeStreetWidth(street.width)
    }
  } else {
    street = _.cloneDeep(undoStack[undoPosition - 1])
  }
  _createDomFromData()
  _segmentsChanged()
  _resizeStreetWidth()

  ignoreStreetChanges = false

  _buildStreetWidthMenu()
  hideAllMenus()

  _saveStreetToServerIfNecessary()
  _saveSettingsLocally()
}

function _propagateUnits () {
  switch (street.units) {
    case SETTINGS_UNITS_IMPERIAL:
      segmentWidthResolution = SEGMENT_WIDTH_RESOLUTION_IMPERIAL
      segmentWidthClickIncrement = SEGMENT_WIDTH_CLICK_INCREMENT_IMPERIAL
      segmentWidthDraggingResolution =
        SEGMENT_WIDTH_DRAGGING_RESOLUTION_IMPERIAL

      document.body.classList.add('units-imperial')
      document.body.classList.remove('units-metric')

      break
    case SETTINGS_UNITS_METRIC:
      segmentWidthResolution = SEGMENT_WIDTH_RESOLUTION_METRIC
      segmentWidthClickIncrement = SEGMENT_WIDTH_CLICK_INCREMENT_METRIC
      segmentWidthDraggingResolution =
        SEGMENT_WIDTH_DRAGGING_RESOLUTION_METRIC

      document.body.classList.add('units-metric')
      document.body.classList.remove('units-imperial')

      break
  }

  _buildStreetWidthMenu()
}
