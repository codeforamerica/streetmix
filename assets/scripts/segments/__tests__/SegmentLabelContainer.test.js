/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import SegmentLabelContainer from '../SegmentLabelContainer'
import { SETTINGS_UNITS_IMPERIAL, SETTINGS_UNITS_METRIC } from '../../users/constants'

const testProps = {
  label: 'foo',
  width: 1,
  units: SETTINGS_UNITS_METRIC,
  locale: 'en'
}

describe('SegmentLabelContainer', () => {
  it('renders string label', () => {
    const wrapper = shallow(<SegmentLabelContainer {...testProps} label="bar" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders React element label', () => {
    const TestComponent = <span>bar</span>
    const wrapper = shallow(<SegmentLabelContainer {...testProps} label={TestComponent} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correct grid styling in metric', () => {
    const wrapper = shallow(<SegmentLabelContainer {...testProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correct grid styling in imperial', () => {
    const wrapper = shallow(<SegmentLabelContainer {...testProps} units={SETTINGS_UNITS_IMPERIAL} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders editable label', () => {
    const wrapper = shallow(<SegmentLabelContainer {...testProps} editable editSegmentLabel={() => {}} />)
    expect(wrapper).toMatchSnapshot()
  })
})
