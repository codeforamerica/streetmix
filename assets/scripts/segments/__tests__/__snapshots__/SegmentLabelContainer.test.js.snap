// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentLabelContainer renders React element label 1`] = `
<div
  className="segment-label-container"
>
  <span
    className="segment-label"
  >
    <span>
      bar
    </span>
  </span>
  <span
    className="segment-width"
  >
    <Memo()
      locale="en"
      units={2}
      value={1}
    />
  </span>
  <span
    className="segment-grid units-metric"
  />
</div>
`;

exports[`SegmentLabelContainer renders correct grid styling in imperial 1`] = `
<div
  className="segment-label-container"
>
  <span
    className="segment-label"
  >
    foo
  </span>
  <span
    className="segment-width"
  >
    <Memo()
      locale="en"
      units={1}
      value={1}
    />
  </span>
  <span
    className="segment-grid units-imperial"
  />
</div>
`;

exports[`SegmentLabelContainer renders correct grid styling in metric 1`] = `
<div
  className="segment-label-container"
>
  <span
    className="segment-label"
  >
    foo
  </span>
  <span
    className="segment-width"
  >
    <Memo()
      locale="en"
      units={2}
      value={1}
    />
  </span>
  <span
    className="segment-grid units-metric"
  />
</div>
`;

exports[`SegmentLabelContainer renders editable label 1`] = `
<div
  className="segment-label-container"
>
  <span
    className="segment-label"
  >
    foo
  </span>
  <span
    className="segment-width"
  >
    <Memo()
      locale="en"
      units={2}
      value={1}
    />
  </span>
  <span
    className="segment-grid units-metric"
  />
</div>
`;

exports[`SegmentLabelContainer renders string label 1`] = `
<div
  className="segment-label-container"
>
  <span
    className="segment-label"
  >
    bar
  </span>
  <span
    className="segment-width"
  >
    <Memo()
      locale="en"
      units={2}
      value={1}
    />
  </span>
  <span
    className="segment-grid units-metric"
  />
</div>
`;
